crates/nu-command/src/filters/chunk_by.rs

```diff
@@ -0,0 +1,220 @@
+use super::utils::chain_error_with_input;
+use nu_engine::{command_prelude::*, ClosureEval};
+use nu_protocol::engine::Closure;
+
+#[derive(Clone)]
+pub struct ChunkBy;
+
+impl Command for ChunkBy {
+    fn name(&self) -> &str {
+        "chunk-by"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("chunk-by")
+            .input_output_types(vec![(
+                Type::List(Box::new(Type::Any)),
+                Type::List(Box::new(Type::Any)),
```

Comment:
This can be a little more specific:
```suggestion
                Type::list(Type::list(Type::Any)),
```